Simulator/Vertices/EventBuffer.h

```diff
@@ -41,7 +36,13 @@ class EventBuffer : public RecordableVector<uint64_t> {
    /// an empty and a full buffer.
    ///
    /// @param maxEvents Defaults to zero; otherwise, buffer size is set
-   EventBuffer(int maxEvents = 0);
+   EventBuffer(int maxEvents = 0)
+   {
+      //RecordableVector<T>();
+      this->dataSeries_.assign(maxEvents, numeric_limits<T>::max());
+      clear();
+      this->setDataType();   // set up data type for recording purpose
```

Comment:
Ditto on the `this` pointer; will not mention for any future uses, but comment applies to them.